cpp/src/slice2cs/Gen.cpp

```diff
@@ -654,6 +640,10 @@ Slice::CsVisitor::getDispatchParams(
 void
 Slice::CsVisitor::emitAttributes(const ContainedPtr& p)
 {
+    assert(
+        dynamic_pointer_cast<Enum>(p->container()) || dynamic_pointer_cast<Enumerator>(p->container()) ||
+        dynamic_pointer_cast<DataMember>(p->container()) || dynamic_pointer_cast<Const>(p->container()));
```

Comment:
This seems bogus, we shouldn't be calling `container` on these. We want to be checking the actual `contained` themselves.

Reply:
Fixed